src/callbacks/analysis_dg2d_parallel.jl

```diff
@@ -0,0 +1,46 @@
+
+function calc_error_norms(func, u::AbstractArray{<:Any,4}, t, analyzer,
+                          mesh::ParallelTreeMesh{2}, equations, initial_condition,
+                          dg::DGSEM, cache)
+  # call the method accepting a general `mesh::TreeMesh{2}`
+  # TODO: MPI, we should improve this; maybe we should dispatch on `u`
+  #       and create some MPI array type, overloading broadcasting and mapreduce etc.
+  #       Then, this specific array type should also work well with DiffEq etc.
+  l2_error, linf_error = invoke(calc_error_norms,
+    Tuple{typeof(func), typeof(u), typeof(t), typeof(analyzer), TreeMesh{2},
+          typeof(equations), typeof(initial_condition), typeof(dg), typeof(cache)},
+    func, u, t, analyzer, mesh, equations, initial_condition, dg, cache)
```

Comment:
I still suggest to implement our own MPI array wrapper and dispatch on that. In this case, we could just call `parent(u)` to get the underlying plain array storage and call that method.

Right now, I think it's premature optimization (for readability) to decide what to do since we don't know our complete MPI story yet and will likely have to re-write lots of code anyway.

Reply:
Fair enough. If you have some ideas about an MPI array wrapper already, should we open an issue for it?

Reply:
Done: #329.

Reply:
Thanks!